python_modules/dagster/dagster/_core/snap/node.py

```diff
@@ -262,6 +263,12 @@ def __new__(
         required_resource_keys: Sequence[str],
         config_field_snap: Optional[ConfigFieldSnap],
     ):
+        # Backcompat for legacy system tags. Older code servers may report the compute kind under
+        # the legacy tag. Code servers running versions after deprecation of the legacy tag may
+        # have both the legacy and current tag set.
+        if LEGACY_COMPUTE_KIND_TAG in tags and COMPUTE_KIND_TAG not in tags:
+            tags = {COMPUTE_KIND_TAG: tags[LEGACY_COMPUTE_KIND_TAG], **tags}
```

Comment:
I think this might cause problematic snapshot id desync between different versions, still investigating